### ISSUE TITLE
Web text input works inconsistently

### ISSUE THREAD

**Issue by brian-superlist** [superlistapp/super_editor#1224]
- Launch the Example Editor with Chrome. (`super_editor/example (branch stable) ❯ flutter run -d chrome`) 
- Type in the document for a bit
- Sometimes, the document updates in response to text input, then eventually stops responding to text input.
- Other times, the editor does not update in response to text input, then somehow starts updating in response to text input.
- May be related to spacebar or delete keys.

Flutter Doctor

```
Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel stable, 3.10.5, on macOS 13.4.1 22F82 darwin-arm64, locale en-US)
    ! The flutter binary is not on your path. Consider adding /Users/brian/lab/flutter/bin to your path.
    ! The dart binary is not on your path. Consider adding /Users/brian/lab/flutter/bin to your path.
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[!] Android Studio (not installed)
[✓] IntelliJ IDEA Community Edition (version 2023.1.3)
[✓] VS Code (version 1.79.2)
[✓] Connected device (3 available)
[✓] Network resources

! Doctor found issues in 2 categories.
```

**Comment by miguelcmedeiros** [comment]
Cc @matthew-carroll

**Comment by matthew-carroll** [comment]
@miguelcmedeiros what's the priority of this for Superlist? Is there a push towards web?

**Comment by matthew-carroll** [comment]
> Sometimes, it works immediately then eventually stops working.

@brian-superlist can you please define "stops working"?

**Comment by brian-superlist** [comment]
Heya @matthew-carroll. What I mean is that:

  - SuperEditor does not respond to text input
  - I can no longer enter text into the document.
  - If I type in letters, the SuperEditor Document does not update.

Here's a video of it happening. The editor works fine the first 15 seconds, then when I type into the editor, the document is not updated:

https://github.com/superlistapp/super_editor/assets/90610075/7090a191-f8ec-4de1-a55e-9690bf9436b2

**Comment by brian-superlist** [comment]
With regards to priority, we are looking to launch our web app soon. We've got the app running against our production environment and are currently working through many bugs. I'd imagine we'll be ready to go on our side in 4 weeks time.

**Comment by angelosilvestre** [comment]
An update:

One of the issues I've found seems to have the same root cause as https://github.com/superlistapp/super_editor/issues/1217, which was fixed recently.

I also found an issue related to new line insertions (https://github.com/superlistapp/super_editor/issues/1219), which I believe it's a Flutter web bug.

I filed https://github.com/flutter/flutter/issues/131023

**Comment by matthew-carroll** [comment]
> One of the issues I've found seems to have the same root cause as https://github.com/superlistapp/super_editor/issues/1217, which was fixed recently.

Are you saying that #1217 wasn't really fixed and there's still a bug?

> I also found an issue related to new line insertions (https://github.com/superlistapp/super_editor/issues/1219), which I believe it's a Flutter web bug.

You're saying that #1219 is caused by a Flutter bug? If so, can you please post the relevant information to that ticket?

**Comment by angelosilvestre** [comment]
@matthew-carroll 

> Are you saying that https://github.com/superlistapp/super_editor/issues/1217 wasn't really fixed and there's still a bug?

No, I was manually testing without the changes from #1217 and I got the same exception. After testing with the changes, I didn't notice the issue described in this ticket anymore. However, with the fix for the paragraph duplication #1219 (our bug seems to be hiding the flutter bug) I noticed the issue mentioned in https://github.com/flutter/flutter/issues/131023.

Even though https://github.com/flutter/flutter/issues/131023 isn't the cause of the original issue described here, it breaks text input on web.

**Comment by matthew-carroll** [comment]
I'm having trouble following that description. Can you summarize what we should know about this ticket at this point?

**Comment by angelosilvestre** [comment]
- This ticket and https://github.com/superlistapp/super_editor/issues/1219 are related.

- The original issue of this ticket seems to be fixed by https://github.com/superlistapp/super_editor/pull/1225. 

- We the fix for https://github.com/superlistapp/super_editor/issues/1219, the assertion failure described in https://github.com/flutter/flutter/issues/131023 is happening, which breaks text input after adding new lines.

**Comment by brian-superlist** [comment]
I've just pulled super_editor `main` branch and run it against Flutter `main` as well. I'm still experiencing the same problems.

https://github.com/superlistapp/super_editor/assets/90610075/5cc0f880-a173-4248-b762-68d6375b0fc8


```
~/superlist/super_editor/super_editor/example  main ❯ git pull             
remote: Enumerating objects: 187, done.
remote: Counting objects: 100% (187/187), done.
remote: Compressing objects: 100% (70/70), done.
remote: Total 187 (delta 128), reused 161 (delta 113), pack-reused 0
Receiving objects: 100% (187/187), 61.37 KiB | 730.00 KiB/s, done.
Resolving deltas: 100% (128/128), completed with 55 local objects.
From github.com:superlistapp/super_editor
   41ba5c4..9a3f57e  1259_backspace-in-empty-node             -> origin/1259_backspace-in-empty-node
 * [new branch]      1259_backspace-in-empty-node_from-stable -> origin/1259_backspace-in-empty-node_from-stable
 * [new branch]      1267_fix-goldens                         -> origin/1267_fix-goldens
   a05c780..9c93480  927_use-content-layers-for-mobile        -> origin/927_use-content-layers-for-mobile
 + 8f5323a...b482bd5 blocknode_update                         -> origin/blocknode_update  (forced update)
   8f6bca1..8699c9d  stable                                   -> origin/stable
Already up to date.

~/superlist/super_editor/super_editor/example  main ❯ ~/lab/flutter/bin/flutter run -d chrome                 

┌─────────────────────────────────────────────────────────┐
│ A new version of Flutter is available!                  │
│                                                         │
│ To update to the latest version, run "flutter upgrade". │
└─────────────────────────────────────────────────────────┘
^C

~/superlist/super_editor/super_editor/example  main ❯ git pull                                                                    
Already up to date.

~/superlist/super_editor/super_editor/example  main ❯ ~/lab/flutter/bin/flutter run -d chrome
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...             15.8s
This app is linked to the debug service: ws://127.0.0.1:65188/xkj0lK09GDo=/ws
Debug service listening on ws://127.0.0.1:65188/xkj0lK09GDo=/ws

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".

A Dart VM Service on Chrome is available at: http://127.0.0.1:65188/xkj0lK09GDo=
The platformViewRegistry getter is deprecated and will be removed in a future release. Please import it from `dart:ui_web` instead.
Initializing logger: ExampleApp
The Flutter DevTools debugger and profiler on Chrome is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:65188/xkj0lK09GDo=
(19.532) ExampleApp > FINE: Showing text format toolbar
(19.548) ExampleApp > FINE: (Re)Building _PositionedToolbar widget due to anchor change
(19.549) ExampleApp > FINE: Anchor is null. Building an empty box.
(19.564) ExampleApp > FINE: (Re)Building _PositionedToolbar widget due to anchor change
(19.564) ExampleApp > FINE: Anchor is non-null: Offset(605.0, 385.3), child: ValueListenableBuilder<DocumentSelection?>
(19.565) ExampleApp > FINE: Building toolbar. Selection: [DocumentSelection] - 
  base: ([DocumentPosition] - node: "cb9d2800-90cf-4d7e-9487-63402456a78d", position: (TextNodePosition(offset: 95, affinity: TextAffinity.downstream))),
  extent: ([DocumentPosition] - node: "cb9d2800-90cf-4d7e-9487-63402456a78d", position: (TextNodePosition(offset: 93, affinity: TextAffinity.downstream)))
(19.690) ExampleApp > FINE: Showing text format toolbar
(19.692) ExampleApp > FINE: Building toolbar. Selection: [DocumentSelection] - 
  base: ([DocumentPosition] - node: "cb9d2800-90cf-4d7e-9487-63402456a78d", position: (TextNodePosition(offset: 95, affinity: TextAffinity.downstream))),
  extent: ([DocumentPosition] - node: "cb9d2800-90cf-4d7e-9487-63402456a78d", position: (TextNodePosition(offset: 88, affinity: TextAffinity.downstream)))
(19.706) ExampleApp > FINE: (Re)Building _PositionedToolbar widget due to anchor change
(19.706) ExampleApp > FINE: Anchor is non-null: Offset(583.5, 385.3), child: ValueListenableBuilder<DocumentSelection?>
(19.813) ExampleApp > FINE: Showing text format toolbar
(19.816) ExampleApp > FINE: Building toolbar. Selection: [DocumentSelection] - 
  base: ([DocumentPosition] - node: "cb9d2800-90cf-4d7e-9487-63402456a78d", position: (TextNodePosition(offset: 95, affinity: TextAffinity.downstream))),
  extent: ([DocumentPosition] - node: "cb9d2800-90cf-4d7e-9487-63402456a78d", position: (TextNodePosition(offset: 81, affinity: TextAffinity.downstream)))
(19.826) ExampleApp > FINE: (Re)Building _PositionedToolbar widget due to anchor change
(19.827) ExampleApp > FINE: Anchor is non-null: Offset(552.0, 385.3), child: ValueListenableBuilder<DocumentSelection?>
(19.930) ExampleApp > FINE: Showing text format toolbar
(19.933) ExampleApp > FINE: Building toolbar. Selection: [DocumentSelection] - 
  base: ([DocumentPosition] - node: "cb9d2800-90cf-4d7e-9487-63402456a78d", position: (TextNodePosition(offset: 95, affinity: TextAffinity.downstream))),
  extent: ([DocumentPosition] - node: "cb9d2800-90cf-4d7e-9487-63402456a78d", position: (TextNodePosition(offset: 76, affinity: TextAffinity.downstream)))
(19.942) ExampleApp > FINE: (Re)Building _PositionedToolbar widget due to anchor change
(19.942) ExampleApp > FINE: Anchor is non-null: Offset(530.3, 385.3), child: ValueListenableBuilder<DocumentSelection?>
(20.181) ExampleApp > FINE: Showing text format toolbar
(20.183) ExampleApp > FINE: Building toolbar. Selection: [DocumentSelection] - 
  base: ([DocumentPosition] - node: "cb9d2800-90cf-4d7e-9487-63402456a78d", position: (TextNodePosition(offset: 95, affinity: TextAffinity.downstream))),
  extent: ([DocumentPosition] - node: "cb9d2800-90cf-4d7e-9487-63402456a78d", position: (TextNodePosition(offset: 70, affinity: TextAffinity.downstream)))
(20.193) ExampleApp > FINE: (Re)Building _PositionedToolbar widget due to anchor change
(20.194) ExampleApp > FINE: Anchor is non-null: Offset(503.9, 385.3), child: ValueListenableBuilder<DocumentSelection?>
(25.284) ExampleApp > FINE: Showing text format toolbar
(25.297) ExampleApp > FINE: (Re)Building _PositionedToolbar widget due to anchor change
(25.297) ExampleApp > FINE: Anchor is null. Building an empty box.
(25.312) ExampleApp > FINE: (Re)Building _PositionedToolbar widget due to anchor change
(25.312) ExampleApp > FINE: Anchor is non-null: Offset(347.1, 385.3), child: ValueListenableBuilder<DocumentSelection?>
(25.313) ExampleApp > FINE: Building toolbar. Selection: [DocumentSelection] - 
  base: ([DocumentPosition] - node: "cb9d2800-90cf-4d7e-9487-63402456a78d", position: (TextNodePosition(offset: 70, affinity: TextAffinity.downstream))),
  extent: ([DocumentPosition] - node: "cb9d2800-90cf-4d7e-9487-63402456a78d", position: (TextNodePosition(offset: 59, affinity: TextAffinity.downstream)))
══╡ EXCEPTION CAUGHT BY SERVICES LIBRARY ╞══════════════════════════════════════════════════════════
The following _Exception was thrown during method call TextInputClient.updateEditingStateWithDeltas:
Exception: Couldn't map an IME position to a document position.
TextEditingValue: 'SuperEditor is a ready-made, configurable document editing '
IME position: TextPosition(offset: 60, affinity: TextAffinity.upstream)

When the exception was thrown, this was the stack:
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 294:49                  throw_
packages/super_editor/src/default_editor/document_ime/document_serialization.dart 271:5       [_imeToDocumentPosition]
packages/super_editor/src/default_editor/document_ime/document_serialization.dart 171:20      imeToDocumentSelection
packages/super_editor/src/default_editor/document_ime/document_delta_editing.dart 320:49      replace
packages/super_editor/src/default_editor/document_ime/document_delta_editing.dart 190:5       [_applyReplacement]
packages/super_editor/src/default_editor/document_ime/document_delta_editing.dart 76:9        applyDeltas
packages/super_editor/src/default_editor/document_ime/document_ime_communication.dart 196:30  updateEditingValueWithDeltas
packages/super_editor/src/default_editor/document_ime/ime_decoration.dart 129:14              updateEditingValueWithDeltas
packages/flutter/src/services/text_input.dart 1864:10                                         _handleTextInputInvocation
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54                            runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 127:5                            _async
packages/flutter/src/services/text_input.dart 1768:45                                         [_handleTextInputInvocation]
packages/flutter/src/services/text_input.dart 1753:20                                         _loudlyHandleTextInputInvocation
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54                            runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 127:5                            _async
packages/flutter/src/services/text_input.dart 1751:51                                         [_loudlyHandleTextInputInvocation]
packages/flutter/src/services/platform_channel.dart 552:55                                    _handleAsMethodCall
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54                            runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 127:5                            _async
packages/flutter/src/services/platform_channel.dart 549:40                                    [_handleAsMethodCall]
packages/flutter/src/services/platform_channel.dart 545:34                                    <fn>
packages/flutter/src/services/binding.dart 567:35                                             <fn>
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54                            runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 127:5                            _async
packages/flutter/src/services/binding.dart 564:98                                             <fn>
lib/_engine/engine/platform_dispatcher.dart 1335:13                                           invoke2
lib/ui/channel_buffers.dart 25:12                                                             invoke
lib/ui/channel_buffers.dart 65:7                                                              push
lib/ui/channel_buffers.dart 131:16                                                            push
lib/_engine/engine/platform_dispatcher.dart 438:25                                            invokeOnPlatformMessage
lib/_engine/engine/text_editing/text_editing.dart 2182:39                                     updateEditingStateWithDelta
lib/_engine/engine/text_editing/text_editing.dart 2289:19                                     <fn>
lib/_engine/engine/text_editing/text_editing.dart 1378:7                                      handleChange
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 574:37              _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 579:39              dcall

call:
  MethodCall(TextInputClient.updateEditingStateWithDeltas, [1, {deltas: [{oldText: SuperEditor is a
  ready-made, configurable document editing a, deltaText: t, deltaStart: 59, deltaEnd: 60,
  selectionBase: 60, selectionExtent: 60, composingBase: -1, composingExtent: -1}]}])
════════════════════════════════════════════════════════════════════════════════════════════════════
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
The platformViewRegistry getter is deprecated and will be removed in a future release. Please import it from `dart:ui_web` instead.
Initializing logger: ExampleApp
══╡ EXCEPTION CAUGHT BY SERVICES LIBRARY ╞══════════════════════════════════════════════════════════
The following _Exception was thrown during method call TextInputClient.updateEditingStateWithDeltas:
Exception: Couldn't map an IME position to a document position.
TextEditingValue: 'SuperEditor is a ready-made, configurable document editing experience. this is a
test of main + '
IME position: TextPosition(offset: 97, affinity: TextAffinity.downstream)

When the exception was thrown, this was the stack:
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 294:49                  throw_
packages/super_editor/src/default_editor/document_ime/document_serialization.dart 271:5       [_imeToDocumentPosition]
packages/super_editor/src/default_editor/document_ime/document_serialization.dart 164:18      imeToDocumentSelection
packages/super_editor/src/default_editor/document_ime/document_delta_editing.dart 227:41      [_applyNonTextChange]
packages/super_editor/src/default_editor/document_ime/document_delta_editing.dart 80:9        applyDeltas
packages/super_editor/src/default_editor/document_ime/document_ime_communication.dart 196:30  updateEditingValueWithDeltas
packages/super_editor/src/default_editor/document_ime/ime_decoration.dart 129:14              updateEditingValueWithDeltas
packages/flutter/src/services/text_input.dart 1864:10                                         _handleTextInputInvocation
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54                            runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 127:5                            _async
packages/flutter/src/services/text_input.dart 1768:45                                         [_handleTextInputInvocation]
packages/flutter/src/services/text_input.dart 1753:20                                         _loudlyHandleTextInputInvocation
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54                            runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 127:5                            _async
packages/flutter/src/services/text_input.dart 1751:51                                         [_loudlyHandleTextInputInvocation]
packages/flutter/src/services/platform_channel.dart 552:55                                    _handleAsMethodCall
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54                            runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 127:5                            _async
packages/flutter/src/services/platform_channel.dart 549:40                                    [_handleAsMethodCall]
packages/flutter/src/services/platform_channel.dart 545:34                                    <fn>
packages/flutter/src/services/binding.dart 567:35                                             <fn>
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54                            runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 127:5                            _async
packages/flutter/src/services/binding.dart 564:98                                             <fn>
lib/_engine/engine/platform_dispatcher.dart 1335:13                                           invoke2
lib/ui/channel_buffers.dart 25:12                                                             invoke
lib/ui/channel_buffers.dart 65:7                                                              push
lib/ui/channel_buffers.dart 131:16                                                            push
lib/_engine/engine/platform_dispatcher.dart 438:25                                            invokeOnPlatformMessage
lib/_engine/engine/text_editing/text_editing.dart 2182:39                                     updateEditingStateWithDelta
lib/_engine/engine/text_editing/text_editing.dart 2289:19                                     <fn>
lib/_engine/engine/text_editing/text_editing.dart 1378:7                                      handleChange
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 574:37              _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 579:39              dcall

call:
  MethodCall(TextInputClient.updateEditingStateWithDeltas, [1, {deltas: [{oldText: SuperEditor is a
  ready-made, configurable document editing experience. this is a test of main + l, deltaText: l,
  deltaStart: 39, deltaEnd: 40, selectionBase: 97, selectionExtent: 97, composingBase: -1,
  composingExtent: -1}]}])
════════════════════════════════════════════════════════════════════════════════════════════════════
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
Another exception was thrown: Exception: Couldn't map an IME position to a document position. 
```

```
~/superlist/super_editor/super_editor/example  main ❯ ~/lab/flutter/bin/flutter doctor       
Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel main, 3.13.0-11.0.pre.7, on macOS 13.4.1 22F82 darwin-arm64, locale en-US)
    ! The flutter binary is not on your path. Consider adding /Users/brian/lab/flutter/bin to your path.
    ! The dart binary is not on your path. Consider adding /Users/brian/lab/flutter/bin to your path.
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[!] Android Studio (not installed)
[✓] IntelliJ IDEA Community Edition (version 2023.1.4)
[✓] VS Code (version 1.80.0)
[✓] Connected device (3 available)
[✓] Network resources

! Doctor found issues in 2 categories.
```

**Comment by angelosilvestre** [comment]
Thanks for the logs. I'll look into this.
